File: app/src/main/java/com/example/valera/homeweatherstation/BuildGraphs/LockableScrollView.java
package com.example.valera.homeweatherstation.BuildGraphs;

/**
 * Created by Valera on 30.03.2015.
 */

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.widget.ScrollView;

public class LockableScrollView extends ScrollView {

    public LockableScrollView(Context context) {
        super(context);
    }

    public LockableScrollView(Context context, AttributeSet attrs) {
        super(context, attrs);

    }

    public LockableScrollView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    private boolean mScrollable = true;

    public void setScrollingEnabled(boolean enabled) {
        mScrollable = enabled;
    }

    public boolean isScrollable() {
        return mScrollable;
    }

    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        switch (ev.getAction()) {
            case MotionEvent.ACTION_DOWN:
                if (mScrollable){
                    return super.onTouchEvent(ev);
                }
                return mScrollable;
            default:
                return super.onTouchEvent(ev);
        }
    }
    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        if (!mScrollable) return false;
        else return super.onInterceptTouchEvent(ev);
    }
}

File: app/src/main/java/me/mervap/ai/home/components/LockableScrollView.kt
package me.mervap.ai.home.components

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.widget.ScrollView

class LockableScrollView : ScrollView {
  constructor(context: Context?) : super(context)
  constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)
  constructor(context: Context?, attrs: AttributeSet?, defStyle: Int) : super(
    context,
    attrs,
    defStyle
  )

  private var isScrollable = true

  fun setScrollingEnabled(enabled: Boolean) {
    isScrollable = enabled
  }

  @SuppressLint("ClickableViewAccessibility")
  override fun onTouchEvent(ev: MotionEvent): Boolean {
    return when (ev.action) {
      MotionEvent.ACTION_DOWN -> {
        if (isScrollable) {
          super.onTouchEvent(ev)
        }
        else {
          false
        }
      }
      else -> super.onTouchEvent(ev)
    }
  }

  override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
    return if (!isScrollable) false else super.onInterceptTouchEvent(ev)
  }
}